causallearn/utils/KCI/KCI.py

```diff
@@ -210,10 +214,18 @@ def get_kappa(self, Kx, Ky):
         _________
         k_appr, theta_appr: approximated parameters of the gamma distribution
 
+        [Updated @Haoyue 06/24/2022]
+        equivalent to:
+            var_appr = 2 * np.trace(Kx.dot(Kx)) * np.trace(Ky.dot(Ky)) / T / T
+        based on the fact that:
+            np.trace(K.dot(K)) == np.sum(K * K.T), where here K is symmetric
+        we can save time on the dot product by only considering the diagonal entries of K.dot(K)
+        time complexity is reduced from O(n^3) (matrix dot) to O(n^2) (traverse each element),
+        where n is usually big (sample size).
         """
         T = Kx.shape[0]
-        mean_appr = np.trace(Kx) * np.trace(Ky) / T
-        var_appr = 2 * np.trace(Kx.dot(Kx)) * np.trace(Ky.dot(Ky)) / T / T
+        mean_appr = np.diag(Kx).sum() * np.diag(Ky).sum() / T # same as np.trace(Kx) * np.trace(Ky) / T. a bit faster
```

Comment:
curious why this will be faster than directly calling np.trace()?

Reply:
Oh my fault. Last time I checked (but maybe with something wrong) and thought that `np.diag(Kx).sum()` is faster by sacrificing the space of new array `np.diag(Kx)` created.

I just double checked. The two's speed performance is almost same. I'll change back to `np.trace`. Thanks!